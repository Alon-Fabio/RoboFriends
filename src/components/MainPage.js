import React, { Component } from 'react';
import CardList from './CardList';
import SearchBox from './SearchBox';
import Scroll from './Scroll';
import ErrorBoundary from './ErrorBoundary'
import './MainPage.css';


class MainPage extends Component {
  constructor() {
    super()
    this.state = {
      robots: []
    }
  }
  

  componentDidMount() {
    this.props.onRequestRobots()
  }

    filteredRobots = robots =>{
        return robots.filter(robot =>{
            return robot.name.toLowerCase().includes(this.props.searchField.toLowerCase());
            })
    }
    // He changed it to â†“, but I think the old fun is better for reuseability.
    // filteredRobots = () =>{
    //     return this.state.robots.filter(robot =>{
    //         return robot.name.toLowerCase().includes(this.props.searchField.toLowerCase());
    //         })
    // }

  render() {
    // const { robots} = this.state;
    const {onSearchChange, robots, isPending} = this.props
    
    return isPending ?
      <h1>Loading</h1> :
      (
        <div className='tc'>
          <h1 className='f1'>RoboFriends</h1>
          <SearchBox searchChange={onSearchChange}/>
          <Scroll>
            <ErrorBoundary>
              <CardList robots={this.filteredRobots(robots)} />
            </ErrorBoundary>
          </Scroll>
        </div>
      );
  }
}

export default MainPage;