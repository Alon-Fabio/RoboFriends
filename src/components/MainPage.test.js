import React from 'react';
import renderer from 'react-test-renderer';
import MainPage from './MainPage';
import {shallow} from 'enzyme';

let wrapper;
let wrapperShallow
beforeEach(()=>{
    let MokeProps = {
        onRequestRobots:jest.fn(),
        robots:[],
        serchField:'',
        isPending:false
    }
    wrapper = renderer.create(<MainPage {...MokeProps}/>).toJSON()
    wrapperShallow = shallow(<MainPage {...MokeProps}/>)
})
describe("MainPage tests", ()=>{
    it("Snapshot test", ()=>{
    
        expect(wrapper).toMatchSnapshot()
    })

    it("Instance tests", ()=>{
        expect(wrapperShallow.instance().filteredRobots([])).toEqual([])
    })
})
