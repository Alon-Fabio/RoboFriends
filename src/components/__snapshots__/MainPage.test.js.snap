// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage tests Snapshot test 1`] = `
<div
  className="tc"
>
  <h1
    className="f1"
  >
    RoboFriends
  </h1>
  <div
    className="pa2"
  >
    <input
      aria-label="Robots search box"
      className="pa3 ba b--green bg-lightest-blue"
      placeholder="search robots"
      type="search"
    />
  </div>
  <div
    style={
      Object {
        "border": "5px solid black",
        "height": "800px",
        "overflow": "scroll",
      }
    }
  >
    <div />
  </div>
</div>
`;
